src/main/java/address/prefs/PrefsManager.java

```diff
@@ -0,0 +1,94 @@
+package address.prefs;
+
+import address.events.EventManager;
+import address.events.MirrorLocationChangedEvent;
+import address.events.SaveLocationChangedEvent;
+
+import java.util.prefs.Preferences;
+import java.io.File;
+
+/**
+ * Manages saving/retrieving of preferences on the user level.
+ * Publicly accessible singleton class.
+ */
+public class PrefsManager {
+
+    public static final String PREF_NODE_NAME_SAVE_LOC = "save-location";
+    public static final String PREF_NODE_NAME_MIRROR_LOC = "mirror-location";
+
+    public static final String DEFAULT_TEMP_FILE_PATH = ".$TEMP_ADDRESS_BOOK";
+
+    private static PrefsManager instance;
+    private static Preferences userPrefs = Preferences.userNodeForPackage(PrefsManager.class);
+
+    public static PrefsManager getInstance(){
+        if (instance == null){
+            instance = new PrefsManager();
+        }
+        return instance;
+    }
+
+    private PrefsManager() {}
+
+    /**
+     * @return the current save file preference or the default temp file if there is no recorded prederence.
+     */
+    public File getSaveLocation() {
+        final String filePath = userPrefs.get(PREF_NODE_NAME_SAVE_LOC, null);
+        return filePath == null ? new File(DEFAULT_TEMP_FILE_PATH) : new File(filePath);
```

Comment:
`filePath` is used in 2 locations on line 38, inlining it would make the ternary statement very long